bitnami/kube-prometheus/values.yaml

```diff
@@ -3307,3 +3307,565 @@ kubeProxy:
+  ##    seccompProfile:
+  ##    type: RuntimeDefault
+  securityContext: {}
+  ## Configure Container Security Context
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container
+  ## @param thanosRuler.containerSecurityContext.enabled Enable container security context
+  ## @param thanosRuler.containerSecurityContext.runAsUser User ID for the container
+  ## @param thanosRuler.containerSecurityContext.runAsNonRoot Enable running as non-root
+  ##
+  containerSecurityContext:
+    enabled: true
+    runAsUser: 1001
+    runAsNonRoot: true
+  ## @param thanosRuler.resourcesPreset Set container resources according to one common preset (allowed values: none, nano, micro, small, medium, large, xlarge, 2xlarge).
+  ## This is ignored if operator.resources is set (operator.resources is recommended for production).
+  ## More information: https://github.com/bitnami/charts/blob/main/bitnami/common/templates/_resources.tpl#L15
+  ##
+  resourcesPreset: "nano"
+  ## @param thanosRuler.resources Set container requests and limits for different resources like CPU or memory (essential for production workloads)
+  ## Example:
+  ## resources:
+  ##   requests:
+  ##     cpu: 2
+  ##     memory: 512Mi
+  ##   limits:
+  ##     cpu: 3
+  ##     memory: 1024Mi
+  ## ref: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
+  ##
+  resources: {}
+  ## @param thanosRuler.containerPorts.http HTTP container port
+  ## @param thanosRuler.containerPorts.grpc GRPC container port
+  ##
+  containerPorts:
+    http: 10902
+    grpc: 10901
+  ## @param thanosRuler.alertQueryUrl The external Query URL the Thanos Ruler will set in the ‘Source’ field of all alerts
+  ## Maps to the ‘–alert.query-url’ CLI arg
+  ##
+  alertQueryUrl: ""
+  ## Configuration for connecting to alertmanager
+  ## Maps to --alertmanagers.config Thanos Ruler argument
+  ## NOTE: This field takes precedence over alertmanagersUrl.
+  ##
+  alertmanagersConfig:
+    ## @param thanosRuler.alertmanagersConfig.existingSecret.name Name of an existing secret to use for Alert Manager config
+    ## If configured, thanosRuler.alertmanagersConfig.config will not be used
+    ## @param thanosRuler.alertmanagersConfig.existingSecret.key Name of a key in the existing secret to use for Alert Manager config
+    ##
+    existingSecret:
+      name: ""
+      key: ""
+    ## @param thanosRuler.alertmanagersConfig.config
+    ## Unused if thanosRuler.alertmanagersConfig.existingSecret.name is configured
+    ## ref: https://thanos.io/tip/components/rule.md/#alertmanager
+    ## e.g:
+    ##   alertmanagers:
+    ##     - http_config:
+    ##         basic_auth:
+    ##           username: some_user
+    ##           password: some_pass
+    ##       static_configs:
+    ##         - alertmanager.thanos.io
+    ##       scheme: http
+    ##       timeout: 10s
+    ##       api_version: v2
+    ##
+    config: {}
```

Comment:
Shouldn't this configuration be auto-generated base on `alertmanager.*` parameters?

Reply:
Definitely. Again, ported from Thanos chart, that didn't include Alertmanager.

I'm thinking of having:
* ability to inject config through existing secret (already in place)
* a default config that uses `alertmanager.*` parameters to point to alertmanager deployed by this chart

What I'm not sure about:
* should we add ability not to have any alertmanager config even if the chart deploys one?
* should we add ability to deploy a static config ignoring the one deployed by the chart? 

Both are edge cases in my mind, typically if you deploy an alertmanager and a ruler with this chart, you will want ruler to talk to that Alertmanager. But maybe there are scenarios and adding support for that will give more flexibility?